apps/mobile/src/components/balance/balance.tsx

```diff
@@ -46,36 +43,42 @@ export function Balance({
   if (isLoading) {
     return <SkeletonLoader height={20} width={100} isLoading={true} />;
   }
-  const hasBalance = balance !== EmptyBalance && typeof balance !== 'string';
-  const isFiat = hasBalance && balance.symbol in currencyNameMap;
-  const balanceSymbol = hasBalance ? balance.symbol : undefined;
-  const maskedCurrencySymbol = !isFiat ? `*${balanceSymbol}` : undefined;
+  if (!balance) {
+    return (
+      <PrivateText color={color} variant={variant}>
+        {EmptyBalanceDisplay}
+      </PrivateText>
+    );
+  }
 
-  const formattedBalance =
-    balance === EmptyBalance ? EmptyBalance : formatBalance({ balance, isFiat, operator });
+  const isFiat = balance && balance.symbol in currencyNameMap;
+  const maskedCurrencySymbol = !isFiat ? `*${balance.symbol}` : undefined;
+
+  const formattedBalance = formatBalance({ balance, isFiat, operator });
 
   if (!lockedBalance) {
     return (
-      <PrivateText mask={maskedCurrencySymbol} color={color} variant={variant}>
-        {formattedBalance}
+      <PrivateText mask={maskedCurrencySymbol} color={color} variant={variant} {...props}>
```

Comment:
I started refactoring this more but we have a potential [new design](https://linear.app/leather-io/issue/LEA-2627/refactor-balance-component-and-show-locked-balances#comment-51764892) now so I will leave it alone and possibly delete it